src/index.js

```diff
@@ -549,9 +550,16 @@ function createConfig(options, entry, format, writeMeta) {
 									...(options.generateTypes !== false && {
 										declarationDir: getDeclarationDir({ options, pkg }),
 									}),
-									jsx: 'preserve',
-									jsxFactory: options.jsx,
-									jsxFragmentFactory: options.jsxFragment,
+									...(options.jsxImportSource
+										? {
+												jsx: 'react-jsx',
+												jsxImportSource: options.jsxImportSource,
+										  }
+										: {
+												jsx: 'preserve',
```

Comment:
should this be `preserve`, or `react`?

Reply:
`preserve` should be correct, though from my quick tests I don't think it makes a difference either way.

`preserve` is to be used when having another tool transpile JSX (e.g., babel), which is what we do. `react` on the other hand has TSC transpile it itself. [TS Docs](https://www.typescriptlang.org/docs/handbook/jsx.html#basic-usage)

I think, at best, `react` would do nothing, and at worst, it'd slow down type generation due to trying to handle JSX which we don't need. No one's raised any issues regarding `preserve` so I'd assume it's fine.